### ISSUE TITLE
Shrink number of shards action in Index Management

### ISSUE THREAD

**Issue by lovelinuxalot** [opendistro-for-elasticsearch/index-management#173]
I would like to place a feature request to add Shrink Index or reduce Number of Shards in the Index Management actions

Just like the Force merge, Read, Replica Count etc. it would be great to have Shrink index, to reduce the number of shards in the actions section. This will be a good feature to have for Managed indices

Currently its really hard to manage the indices as they are managed with setting the number of shards at the creation of index and only increasing the number of replicas. I am currently using managed index only for small indices. For larger indices, the indices are managed outside of Index Management by creating new weekly index with less number of shards and reindex all the daily indices in the week into the weekly index

Thanks

**Comment by yc1024** [comment]
@dbbaughe Is there a release plan for this PR? I look forward to this action.

**Comment by dbbaughe** [comment]
Hi @yc1024, it's definitely on our backlog of things to add, but as of right now it hasn't been scheduled yet. Will update this issue when the status changes.

**Comment by yc1024** [comment]
@dbbaughe Thanks for your reply.

**Comment by arnitolog** [comment]
@dbbaughe do you guys have this feature somewhere in your roadmap?

**Comment by Cthulhunonamous** [comment]
+1, many valid use cases for this feature

**Comment by jsirianni** [comment]
This would be useful. It is nice to reduce the shard count after moving an index to "warm" / "cold", for workloads that require a large shard count for writes but not for reads / long term storage.

**Comment by quad2524** [comment]
+1 , same issue with needing many shards for ingest and then shrink down for reading.

**Comment by Thoro** [comment]
This is actually stopping me from migrating from my custom solution to ISM - shrink is a supported action in ES ILM

**Comment by quad2524** [comment]
> This is actually stopping me from migrating from my custom solution to ISM - shrink is a supported action in ES ILM

I implemented using elastic/curator as a stop gap here to shrink my 10 shard indexes down to 1 when the filter is met and it is working okay for me. 
```
    options:
      shrink_node: DETERMINISTIC
      ignore_empty_list: True
      disable_action: False
      number_of_shards: 1
      wait_for_completion: true
      node_filters:
        permit_masters: True
```

**Comment by ansmathu** [comment]
We picked up this issue and will be working on it soon.

**Comment by tumblingtumbleweed** [comment]
Super important feature here. Any idea when it will be scheduled for release?

**Comment by arnitolog** [comment]
Guys, any updates on the release date for this feature?

**Comment by alexmaurizio** [comment]
+1. I was used to have this feature in ES ILM and I think this feature is very important here too.

**Comment by jianchen2580** [comment]
+1

**Comment by pmarjou22** [comment]
+1

**Comment by downsrob** [comment]
This feature is about to be released on May 24th as a part of OpenSearch 2.0! You can see this issue mirrored in the OpenSearch index management repository [here](https://github.com/opensearch-project/index-management/issues/40). All of the pull requests are attached to the issue.
